sql/src/main/java/io/crate/expression/symbol/SymbolVisitors.java

```diff
@@ -42,6 +48,127 @@ public static boolean any(Predicate<? super Symbol> symbolPredicate, Symbol symb
         return symbol.accept(ANY_VISITOR, symbolPredicate);
     }
 
+    /**
+     * Calls the given `consumer` for all intersection points between `needle` and `haystack`.
+     * For example, in:
+     * <pre>
+     *     needle: x > 20 AND y = 2
+     *     haystack: [x, y = 2]
+     * </pre>
+     *
+     * The `consumer` would be called for `x` and for `y = 2`
+     */
+    public static <T> void intersection(Symbol needle, Collection<T> haystack, Consumer<T> consumer) {
+        needle.accept(new IntersectionVisitor<>(haystack, consumer), null);
```

Comment:
minor: was wondering whether it is really necessary to create an instance of `IntersectionVisitor` on each `intersection` method invocation and whether it makes sense to reuse a visitor context to pass haystack and needle?

Reply:
> but it is probably the same coz the context has to be created..

Yes, that was my reason for going for a visitor instance directly - to avoid having to introduce a context class.